config/software/datadog-trace-agent.rb

```diff
@@ -0,0 +1,35 @@
+name "datadog-trace-agent"
+default_version "master"
+
+gourl = "https://storage.googleapis.com/golang/go1.7.1.linux-amd64.tar.gz"
+goout = "go.tar.gz"
+godir = "/usr/local/go17"
+gobin = "#{godir}/go/bin/go"
+
+env = {
+  "GOPATH" => "#{Omnibus::Config.cache_dir}/src/#{name}",
+  "GOROOT" => "/usr/local/go17/go",
+  "PATH" => "#{godir}/go/bin:#{ENV["PATH"]}"
+}
+
+build do
+   ship_license "https://raw.githubusercontent.com/DataDog/datadog-trace-agent/#{version}/LICENSE"
+   # download go1.7
+   command "curl #{gourl} -o #{goout}"
+   command "mkdir -p #{godir}"
+   command "tar zxfv #{goout} -C #{godir}"
+
+   # Download datadog-trace-agent without installing
+   command "git clone https://github.com/DataDog/datadog-trace-agent.git $GOPATH/src/github.com/DataDog/datadog-trace-agent", :env => env
+   # Checkout datadog-trace-agent's build dependencies
+   command "#{gobin} get -d github.com/robfig/glock", :env => env
+
+   # Pin build deps to known versions
+   command "git checkout 7bc8ce51048e2adc11733f90a87b1c02fb7feebe", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/robfig/glock"
+   command "#{gobin} install github.com/robfig/glock", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/robfig/glock"
+
+   # Checkout and build datadog-trace-agent
+   command "$GOPATH/bin/glock sync github.com/DataDog/datadog-trace-agent", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/DataDog/datadog-trace-agent"
+   command "git checkout master && git pull", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/DataDog/datadog-trace-agent"
```

Comment:
i don't think you should always use master and you shouldn't do the checkout yourself you should let omnibus does that for you and have a proper tagging scheme in place for the datadog-trace-agent repo (similar to the dd-agent one)